pkg/cmd/repo/garden/garden.go

```diff
@@ -0,0 +1,412 @@
+		Hidden: true,
+		RunE: func(c *cobra.Command, args []string) error {
+			if len(args) > 0 {
+				opts.RepoArg = args[0]
+			}
+			if runF != nil {
+				return runF(&opts)
+			}
+			return gardenRun(&opts)
+		},
+	}
+
+	return cmd
+}
+
+func gardenRun(opts *GardenOptions) error {
+	out := opts.IO.Out
+
+	if runtime.GOOS == "windows" {
+		return errors.New("sorry :( this command only works on linux and macos")
+	}
+
+	if !opts.IO.IsStdoutTTY() {
+		return errors.New("must be connected to a terminal")
+	}
+
+	httpClient, err := opts.HttpClient()
+	if err != nil {
+		return err
+	}
+
+	var toView ghrepo.Interface
+	apiClient := api.NewClientFromHTTP(httpClient)
+	if opts.RepoArg == "" {
+		var err error
+		toView, err = opts.BaseRepo()
+		if err != nil {
+			return err
+		}
+	} else {
+		var err error
+		viewURL := opts.RepoArg
+		if !strings.Contains(viewURL, "/") {
+			currentUser, err := api.CurrentLoginName(apiClient, ghinstance.Default())
+			if err != nil {
+				return err
+			}
+			viewURL = currentUser + "/" + viewURL
+		}
+		toView, err = ghrepo.FromFullName(viewURL)
+		if err != nil {
+			return fmt.Errorf("argument error: %w", err)
+		}
+	}
+
+	seed := computeSeed(ghrepo.FullName(toView))
+	rand.Seed(seed)
+
+	termWidth, termHeight, err := utils.TerminalSize(out)
+	if err != nil {
+		return err
+	}
+
+	termWidth -= 10
+	termHeight -= 10
+
+	geo := &Geometry{
+		Width:      termWidth,
+		Height:     termHeight,
+		Repository: toView,
+		// TODO based on number of commits/cells instead of just hardcoding
+		Density: 0.3,
+	}
+
+	maxCommits := geo.Width * geo.Height
+
+	commits, err := getCommits(httpClient, toView, maxCommits)
+	if err != nil {
+		return err
+	}
+	player := &Player{0, 0, utils.Bold("@"), geo, 0}
+
+	garden := plantGarden(commits, geo)
+	clear(opts.IO)
+	drawGarden(out, garden, player)
+
+	// thanks stackoverflow https://stackoverflow.com/a/17278776
+	if runtime.GOOS == "darwin" {
+		_ = exec.Command("stty", "-f", "/dev/tty", "cbreak", "min", "1").Run()
+		_ = exec.Command("stty", "-f", "/dev/tty", "-echo").Run()
+	} else {
+		_ = exec.Command("stty", "-F", "/dev/tty", "cbreak", "min", "1").Run()
+		_ = exec.Command("stty", "-F", "/dev/tty", "-echo").Run()
+	}
+
+	var b []byte = make([]byte, 1)
+	for {
+		_, err := opts.IO.In.Read(b)
+		if err != nil {
+			return err
+		}
+
+		quitting := false
+		switch {
+		case isLeft(b):
+			player.move(DirLeft)
+		case isRight(b):
+			player.move(DirRight)
+		case isUp(b):
+			player.move(DirUp)
+		case isDown(b):
+			player.move(DirDown)
+		case isQuit(b):
+			quitting = true
+		}
+
+		if quitting {
+			break
+		}
+
+		clear(opts.IO)
+		drawGarden(out, garden, player)
+	}
+
+	fmt.Println()
+	fmt.Println(utils.Bold("You turn and walk away from the wildflower garden..."))
+
+	return nil
+}
+
+// TODO fix arrow keys
+
+func isLeft(b []byte) bool {
+	return bytes.EqualFold(b, []byte("a")) || bytes.EqualFold(b, []byte("q")) || bytes.EqualFold(b, []byte("h"))
```

Comment:
If <kbd>q</kbd> is Left now, how do we exit the garden?

Reply:
The only way I can exit (on macOS at least) seems to be <kbd>CTRL</kbd> <kbd>C</kbd>. 
That seems acceptable / consistent with `gh credits`? `Q` doesn't work for me there either FWIW.

https://github.com/cli/cli/blob/5e11fe8fa0851006fdd894e4949abb722450708e/pkg/cmd/repo/credits/credits.go#L190

Reply:
just switching to ctrl+c for quitting seems ok to me

Reply:
Sure! I was simply asking about `q` since it was documented as an exit technique for the garden (<kbd>q</kbd> in `gh credits` was never intended to work)

<kbd>Ctrl-C</kbd> works, but does not print the nice exit message:
> You turn and walk away from the wildflower garden…

Reply:
After testing this for a while I was unhappy about losing q; I ended up just making the arrow keys work, finally, so that is an option for anyone who doesn't want WADS or vi keys.